src/raps/registerENS.ts

```diff
@@ -163,16 +182,29 @@ export const createTransferENSRap = async (
       !isEmpty(emptyRecords) &&
       recordsForTransactionAreValid(ensRegistrationRecords);
     if (validRecords) {
-      const shouldUseMulticall = shouldUseMulticallTransaction(
-        ensRegistrationRecords
-      );
-      const recordsAction = createNewENSAction(
-        shouldUseMulticall
-          ? RapActionTypes.multicallENS
-          : RapActionTypes.setTextENS,
-        { ...ensActionParameters, records: newRecords }
-      );
-      actions = concat(actions, recordsAction);
+      const txType = getTransactionTypeForRecords(ensRegistrationRecords);
+      let rapActionType;
+      switch (txType) {
+        case ENSRegistrationTransactionType.MULTICALL:
+          rapActionType = RapActionTypes.multicallENS;
+          break;
+        case ENSRegistrationTransactionType.SET_ADDR:
+          rapActionType = RapActionTypes.setAddrENS;
+          break;
+        case ENSRegistrationTransactionType.SET_TEXT:
+          rapActionType = RapActionTypes.setTextENS;
+          break;
+        default:
+          rapActionType = null;
+          break;
+      }
```

Comment:
this switch is in the code about 3 times already, could we create another helper to get the rapActionType directly?